python/lambda-ddb-mysql-etl-pipeline/lambda/asteroids.py

```diff
@@ -0,0 +1,155 @@
+import requests
+import json
+from requests.exceptions import Timeout
+from requests.exceptions import HTTPError
+from botocore.exceptions import ClientError
+from datetime import date
+import csv
+import os
+import boto3
+import logging
+
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+
+class Asteroids:
+    """Client to NASA API and execution interface to branch data processing by file type.
+
+    Notes:
+        This class doesn't look like a normal class. It is a simple example of how one might
+        workaround AWS Lambda's limitations of class use in handlers. It also allows for 
+        better organization of code to simplify this example. If one planned to add
+        other NASA endpoints or process larger amounts of Asteroid data for both .csv and .json formats,
+        asteroids_json and asteroids_csv should be modularized and divided into separate lambdas
+        where stepfunction orchestration is implemented for a more comprehensive workflow.
+        However, for the sake of this demo I'm keeping it lean and easy.
+    """
+
+    def execute(self, format):
+        """Serves as logical interface to assign class attributes and execute class methods
+
+        Raises:
+            Exception: If file format is not of .json or .csv file types.
+        Notes:
+            Have fun!
+        """
+        self.file_format=format
+        self.today=date.today().strftime('%Y-%m-%d')
+        # method call below used when Secrets Manager integrated. See get_secret.__doc__ for more.
+        # self.api_key=get_secret('nasa_api_key')
+        self.api_key=os.environ["NASA_KEY"]
+        self.endpoint=f"""https://api.nasa.gov/neo/rest/v1/feed?start_date={self.today}&end_date={self.today}&api_key={self.api_key}"""
+        self.response_object=self.nasa_client(self.endpoint)
+        self.processed_response=self.process_asteroids(self.response_object)
+        if self.file_format == "json":
+            self.asteroids_json(self.processed_response)
+        elif self.file_format == "csv":
+            self.asteroids_csv(self.processed_response)
+        else:
+            raise Exception("FILE FORMAT NOT RECOGNIZED")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.